src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1108,6 +1109,51 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfy(actual, unsatisfiedRequirements, info));
   }
 
+  /**
+   * Verifies that the all the given consumers can be satisfied by elements in the iterable with
+   * an element at most satisfies one consumer. No order requirement.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Iterable}.
+   * @param consumers the consumers that are expected to be satisfied by the elements of the given {@code Iterable}.
+   * @throws NullPointerException if the given consumers is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if any {@code Consumer} in the consumers cannot be satisfied by elements in the given {@code Iterable}.
+   */
+  @SafeVarargs
+  public final <E> void assertSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E>... consumers) {
+    assertNotNull(info, actual);
+    requireNonNull(consumers, "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    requireNonNull(consumers[0], "The Consumer<? super E>... expressing the assertions consumers must not be null");
```

Comment:
Is there a particular reason why the first consumer must be non-null? If so, why is it necessary that the consumers is non-empty (and is the length requirement checked somewhere else)? Should other consumers also be explicitly checked for non-nullness?

Reply:
"Is there a particular reason why the first consumer (`consumers[0]`) must be non-null?": Oh, it's my mistake! Thanks! 
"why is it necessary that the `consumers` is non-empty": Maybe the test can be directly passed if the `consumers` is empty or null?

Reply:
Could it be that the wrong line (```consumers``` instead of ```consumers[0]```) was accidentally removed?

Reply:
I am thinking about: if a consumer is empty or null, what will happen?